### PR TITLE
 [12.0][MIG] keychain - WIP

### PR DESCRIPTION
**Pull request by hparfr**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Hi, Raph, can you Travis and runbot status?

**Comment by hparfr** [comment]
OCB branch is good now, thanks Pedro

**Comment by pedrobaeza** [comment]
Can you please attend these warnings (it's a new Odoo check):

```
2018-10-01 13:48:02,744 150 WARNING openerp_test odoo.models: The model keychain.account has no _description
2018-10-01 13:48:02,745 150 WARNING openerp_test odoo.models: The model keychain.backend has no _description
```

**Comment by rvalyi** [comment]
so if I'am right, this PR is superseded by this one https://github.com/OCA/server-auth/pull/39, right?
If so, may be we should close this PR and mention the keychain origin in the new data_encryption module. cc @hparfr @florian-dacosta

File: keychain/models/keychain.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    def _compute_password(self):
+        # Only needed in v8 for _description_searchable issues
+        return True
+
+    def _get_password(self):
+        """Password in clear text."""
+        try:
+            return self._decode_password(self.password)
+        except Warning as warn:
+            raise Warning(_(
```
Comment (rousseldenis) [inline keychain/models/keychain.py:70]: IMHO this is incorrect to raise not ORM exceptions.

File: keychain/models/keychain.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        inverse='_inverse_set_password',
+        compute='_compute_password',
+        store=False)
+    password = fields.Text(
+        help="Password is derived from clear_password",
+        readonly=True)
+    data = fields.Text(help="Additionnal data as json")
+
+    def _compute_password(self):
+        # Only needed in v8 for _description_searchable issues
```
Comment (rousseldenis) [inline keychain/models/keychain.py:62]: I think this can be removed

File: keychain/models/keychain.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        """
+        domain.append(['environment', 'in', self._retrieve_env()])
+        return self.search(domain)
+
+    @api.multi
+    def write(self, vals):
+        """At this time there is no namespace set."""
+        if not vals.get('data') and not self.data:
+            vals['data'] = self._serialize_data(self._init_data())
+        return super(KeychainAccount, self).write(vals)
```
Comment (rousseldenis) [inline keychain/models/keychain.py:115]: Can be simplified

```suggestion
        return super().write(vals)
```

**Review by rousseldenis — COMMENTED** [review]
LGTM. Some comments

@hparfr Could we try to move forward on this one ?

**Comment by florian-dacosta** [comment]
@rousseldenis 
Maybe we should close this PR.
Indeed, we have refactored the module and would use : https://github.com/OCA/server-env/pull/24
which is based on server_environment
So, I don't think we will continue this PR.

**Comment by rousseldenis** [comment]
@florian-dacosta Ok, thanks.

@hparfr Could you close ?

**Pull request closed without merging**